lib/core/test/unit/Cardano/Wallet/Api/ServerSpec.hs

```diff
@@ -265,3 +324,140 @@ errorHandlingSpec = describe "liftHandler and toServerError" $ do
 
         runHandler (getNetworkClock (error "bomb") True)
             `shouldReturn` Left expectedErr
+
+listStakeKeysSpec :: Spec
+listStakeKeysSpec = do
+    describe "Listing stake keys" $ do
+        it "no rewards => sum of all keys' stake == utxo balance" $ do
+            property prop_listStakeKeysBalance
+        it "ours and foreign keys are disjoint" $ do
+            property prop_listStakeKeysDisjoint
+
+prop_listStakeKeysDisjoint
+    :: PrettyShow UTxO
+    -> SaneOurStakeKeysInput
+    -> (Map RewardAccount Coin)
+    -> Property
+prop_listStakeKeysDisjoint (PrettyShow utxo) (SaneOurStakeKeysInput ours) m =
+    let
```

Comment:
Since there's already a `where`, you don't need `let`.

Reply:
For readability I think it makes sense to have `let .. in ..` be meant to be read sequentially, and keep auxiliary helpers in `where` clauses. The `res` is very important, but `disjoint` is not.